gsonpath-compiler/src/main/java/gsonpath/generator/standard/ReadFunctions.kt

```diff
@@ -275,7 +281,11 @@ private fun isCheckIfNullApplicable(gsonField: GsonField, requiresConstructorInj
 /**
  * Writes the Java code for field reading that is not supported by Gson.
  */
-private fun writeGsonFieldReading(codeBlock: CodeBlock.Builder, gsonField: GsonField, requiresConstructorInjection: Boolean): FieldReaderResult {
+private fun writeGsonFieldReading(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=kotlin%3AS3776)